src/pages/register/index.js

```diff
@@ -8,12 +8,42 @@ import "./register.css";
 const Register = () => {
 	const { onRegister } = useAuth();
 	const [formData, setFormData] = useState({ email: "", password: "" });
+	const [isEmailValid, setIsEmailValid] = useState(null)
+	const [isPasswordValid, setIsPasswordValid] = useState(null)
+	const [isEmailTaken, setIsEmailTaken] = useState(null)
 
 	const onChange = (e) => {
 		const { name, value } = e.target;
 		setFormData({ ...formData, [name]: value });
 	};
 
+	const handelSubmit = () => {
+		setIsEmailTaken(false)
+
+		const validateEmail = (email) => {
+			if (email.length > 254) return false
+			
+			const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+			return regex.test(email)
+		}
+
+		const validatePassword = (password) => {
+			const regex = /^(?=.*[\d])(?=.*[!@#$%^&*])[\w!@#$%^&*]{8,}$/
+			return regex.test(password)
+		}
```

Comment:
These functions will be useful for the login for as well, I'd move these to a different file where any other component can use them. Perhaps `utils` folder with a `validation.js` file inside?

Reply:
Should I then create a new folder in `src` called `utils`, don't see one already set up.

Reply:
Yeah, you can create a utils folder for this